documentation/source/Checkpoints.md

```diff
@@ -0,0 +1,282 @@
+
+   ```python
+        from super_gradients.training import models
+        from super_gradients.common.object_names import Models
+        from super_gradients.training.utils.checkpoint_utils import load_checkpoint_to_model
+        
+        model = models.get(model_name=Models.RESNET18, num_classes=10)
+        load_checkpoint_to_model(net=model, ckpt_local_path="/path/to/my_checkpoints_folder/my_resnet18_training_experiment/ckpt_best.pth")
+   ```
+### Extending the Functionality of PyTorch's `strict` Parameter in `load_state_dict()`
+
+When not familiar with PyTorch's `strict` parameter in `load_state_dict()`, please see [PyTorch's docs on this matter](https://pytorch.org/tutorials/beginner/saving_loading_models.html#id4) first.
+
+The equivalent arguments for PyTorch's `strict` parameter in `load_state_dict()` in `models.get()` and `load_checkpoint_to_model` are `strict` and `strict_load` respectively, and expect SG's `StrictLoad` enum type.
+
+Let's have a look at its possible values:
+
+```python
+
+class StrictLoad(Enum):
+    """
+    Wrapper for adding more functionality to torch's strict_load parameter in load_state_dict().
+    Attributes:
+        OFF              - Native torch "strict_load = off" behavior. See nn.Module.load_state_dict() documentation for more details.
+        ON               - Native torch "strict_load = on" behavior. See nn.Module.load_state_dict() documentation for more details.
+        NO_KEY_MATCHING  - Allows the usage of SuperGradient's adapt_checkpoint function, which loads a checkpoint by matching each
+                           layer's shapes (and bypasses the strict matching of the names of each layer (i.e., disregards the state_dict key matching)).
+    """
+
+    OFF = False
+    ON = True
+    NO_KEY_MATCHING = "no_key_matching"
+
+```
+
+In other words, we added another loading mode option- `no_key_matching`. This option exploits the fact that the `state_dicts` are `OrderedDict`s, and comes in handy when the underlying network's structure remains the same, but the `state_dict`s keys do not match the ones inside the models `state_dict`.
+Let's demonstrate the different strict modes with a simple example:
+
+```python
+import torch
+
+class ModelA(torch.nn.Module):
+    def __init__(self):
+        super(ModelA, self).__init__()
+        self.conv1 = torch.nn.Conv2d(3, 6, 5)
+        self.conv2 = torch.nn.Conv2d(6, 16, 5)
+
+class ModelB(torch.nn.Module):
+    def __init__(self):
+        super(ModelB, self).__init__()
+        self.conv1 = torch.nn.Conv2d(3, 6, 5)
+        self.CONV2 = torch.nn.Sequential([torch.nn.Conv2d(6, 16, 5)])
+        
+```
+
+Notice the above networks have identical weight structures but will have different keys in their `state_dict`s.
+This is why loading a checkpoint from either one to the other, using `strict=True`, will fail and crash. Using `strict=True` will not crash and successfully load the first layer's weights only.
```

Comment:
the second one should be `strict=false`

Reply:
fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422